Submitter bump fee mechanism bugs

Version: `v0.19.0`

Relevant configuration:
```shell
    submitter:
      resubmit-fee-multiplier: 1.3
      polling-interval-seconds: 60
      resend-interval-seconds: 1200
    btc:
      estimate-mode: CONSERVATIVE
      tx-fee-min: 1000
      tx-fee-max: 10000
      default-fee: 1000
      target-block-num: 6
```

Bump fee behaviour: https://mempool.space/signet/tx/511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349

Log from the latest replacemenets:

```shell
2025-01-03T01:26:58.285091Z	info	Polling sealed raw checkpoints...	{"module": "submitter"}
2025-01-03T01:26:58.453789Z	debug	Next polling happens in 60 seconds	{"module": "submitter"}
2025-01-03T01:26:58.453821Z	info	A sealed raw checkpoint for epoch 216 is found	{"module": "submitter"}
2025-01-03T01:26:58.453829Z	debug	The checkpoint for epoch 216 was sent more than 1200 seconds ago but not included on BTC	{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.453838Z	debug	current minimum relay fee rate is 1012 sat/kb	{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.453853Z	debug	the bumped fee: 0.00011768 BTC Satoshis, the required fee: 0.00009209 BTC Satoshis	{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.453861Z	debug	Resending the second tx of the checkpoint 216, old fee of the second tx: 0.00009052 BTC Satoshis, txid: edca4553d7642ddafb27fbe161a1f2904220aa00365f84d88812e307232121c5{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.563394Z	debug	Sending tx 511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349 to BTC	{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.566269Z	debug	Successfully sent tx 511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349 to BTC	{"module": "submitter", "module": "relayer"}
2025-01-03T01:26:58.566327Z	info	Successfully re-sent the second tx of the checkpoint 216, txid: 511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349, bumped fee: 0.00011768 BTC Satoshis	{"module": "submitter", "module": "relayer"}
2025-01-03T01:27:58.285106Z	info	Polling sealed raw checkpoints...	{"module": "submitter"}
2025-01-03T01:27:58.393824Z	debug	Next polling happens in 60 seconds	{"module": "submitter"}
2025-01-03T01:27:58.393847Z	info	A sealed raw checkpoint for epoch 216 is found	{"module": "submitter"}
2025-01-03T01:27:58.394025Z	debug	The checkpoint for epoch 216 was sent more than 1200 seconds ago but not included on BTC	{"module": "submitter", "module": "relayer"}
2025-01-03T01:27:58.394059Z	debug	current minimum relay fee rate is 1012 sat/kb	{"module": "submitter", "module": "relayer"}
2025-01-03T01:27:58.394143Z	debug	the bumped fee: 0.00015298 BTC Satoshis, the required fee: 0.00011925 BTC Satoshis	{"module": "submitter", "module": "relayer"}
2025-01-03T01:27:58.394172Z	debug	Resending the second tx of the checkpoint 216, old fee of the second tx: 0.00011768 BTC Satoshis, txid: 511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349{"module": "submitter", "module": "relayer"}
2025-01-03T01:27:58.423959Z	debug	Transaction 511393a0695cab68403acfd36b4461ca106805fd1d49a9f85021468b87a66349 is already confirmed	{"module": "submitter", "module": "relayer"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x30ae7dd]

goroutine 59 [running]:
github.com/babylonlabs-io/vigilante/submitter/relayer.(*Relayer).MaybeResubmitSecondCheckpointTx(0xc0005821b0, 0xc001c20640?)
	/go/src/github.com/babylonlabs-io/vigilante/submitter/relayer/relayer.go:209 +0x45d
github.com/babylonlabs-io/vigilante/submitter.(*Submitter).processCheckpoints(0xc001317450)
	/go/src/github.com/babylonlabs-io/vigilante/submitter/submitter.go:217 +0x2a5
created by github.com/babylonlabs-io/vigilante/submitter.(*Submitter).Start in goroutine 1
	/go/src/github.com/babylonlabs-io/vigilante/submitter/submitter.go:133 +0x199
```

Issues:
- We exceeded the `tx-fee-max`
- After the initial fee bump, we bump again the fee every `polling-interval-seconds` instead of every `resend-interval-seconds`
- Service panic

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -37,6 +37,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 
 ## Unreleased
 
+### Bug Fixes
+
+* [#154](https://github.com/babylonlabs-io/vigilante/pull/154) fix: panic in maybeResendSecondTxOfCheckpointToBTC
+
 ## v0.19.0
 
 ### Bug Fixes

--- a/config/grpc.go
+++ b/config/grpc.go
@@ -1,10 +1,5 @@
 package config
 
-const (
-	// DefaultGRPCAddress defines the default address to bind the gRPC server to.
-	DefaultGRPCAddress = "0.0.0.0:8080"
-)
-
 // GRPCConfig defines configuration for the gRPC server.
 type GRPCConfig struct {
 	OneTimeTLSKey bool     `mapstructure:"onetime-tls-key"`

--- a/config/grpcweb.go
+++ b/config/grpcweb.go
@@ -1,10 +1,5 @@
 package config
 
-const (
-	// DefaultGRPCWebAddress defines the default address to bind the gRPC-web server to.
-	DefaultGRPCWebAddress = "0.0.0.0:8081"
-)
-
 // GRPCWebConfig defines configuration for the gRPC-web server.
 type GRPCWebConfig struct {
 	Placeholder string `mapstructure:"placeholder"`

--- a/submitter/relayer/relayer.go
+++ b/submitter/relayer/relayer.go
@@ -192,14 +192,18 @@ func (rl *Relayer) MaybeResubmitSecondCheckpointTx(ckpt *ckpttypes.RawCheckpoint
 		return nil
 	}
 
-	rl.logger.Debugf("Resending the second tx of the checkpoint %v, old fee of the second tx: %v Satoshis, txid: %s",
+	rl.logger.Debugf("Maybe resending the second tx of the checkpoint %v, old fee of the second tx: %v Satoshis, txid: %s",
 		ckptEpoch, rl.lastSubmittedCheckpoint.Tx2.Fee, rl.lastSubmittedCheckpoint.Tx2.TxID.String())
 
-	resubmittedTx2, err := rl.resendSecondTxOfCheckpointToBTC(rl.lastSubmittedCheckpoint.Tx2, bumpedFee)
+	resubmittedTx2, err := rl.maybeResendSecondTxOfCheckpointToBTC(rl.lastSubmittedCheckpoint.Tx2, bumpedFee)
 	if err != nil {
 		rl.metrics.FailedResentCheckpointsCounter.Inc()
 
 		return fmt.Errorf("failed to re-send the second tx of the checkpoint %v: %w", rl.lastSubmittedCheckpoint.Epoch, err)
+	}
+
+	if resubmittedTx2 == nil {
+		return nil
 	}
 
 	// record the metrics of the resent tx2
@@ -255,17 +259,16 @@ func (rl *Relayer) calculateBumpedFee(ckptInfo *types.CheckpointInfo) btcutil.Am
 	return ckptInfo.Tx2.Fee.MulF64(rl.config.ResubmitFeeMultiplier)
 }
 
-// resendSecondTxOfCheckpointToBTC resends the second tx of the checkpoint with bumpedFee
-func (rl *Relayer) resendSecondTxOfCheckpointToBTC(tx2 *types.BtcTxInfo, bumpedFee btcutil.Amount) (*types.BtcTxInfo, error) {
-	_, status, err := rl.TxDetails(rl.lastSubmittedCheckpoint.Tx2.TxID,
-		rl.lastSubmittedCheckpoint.Tx2.Tx.TxOut[changePosition].PkScript)
+// maybeResendSecondTxOfCheckpointToBTC resends the second tx of the checkpoint with bumpedFee
+func (rl *Relayer) maybeResendSecondTxOfCheckpointToBTC(tx2 *types.BtcTxInfo, bumpedFee btcutil.Amount) (*types.BtcTxInfo, error) {
+	_, status, err := rl.TxDetails(tx2.TxID, tx2.Tx.TxOut[changePosition].PkScript)
 	if err != nil {
 		return nil, err
 	}
 
 	// No need to resend, transaction already confirmed
 	if status == btcclient.TxInChain {
-		rl.logger.Debugf("Transaction %v is already confirmed", rl.lastSubmittedCheckpoint.Tx2.TxID)
+		rl.logger.Debugf("Transaction %v is already confirmed", tx2.TxID)
 
 		return nil, nil
 	}

